openadapt/privacy/providers/private_ai.py

```diff
@@ -0,0 +1,184 @@
+            is_separated (bool): Whether the text is separated with special characters
+
+        Returns:
+            str: Scrubbed text
+        """
+        url = "https://api.private-ai.com/deid/v3/process/text"
+
+        payload = {
+            "text": [text],
+            "link_batch": False,
+            "entity_detection": {
+                "accuracy": "high",
+                "return_entity": True,
+            },
+            "processed_text": {
+                "type": "MARKER",
+                "pattern": "[UNIQUE_NUMBERED_ENTITY_TYPE]",
+            },
+        }
+
+        headers = {
+            "Content-Type": "application/json",
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(url, json=payload, headers=headers)
+        if response is None:
+            raise ValueError("Private AI request returned None")
+
+        data = response.json()
+        logger.debug(data)
+        if type(data) is dict and "detail" in data.keys():
+            raise ValueError(data.get("detail"))
+
+        redacted_text = data[0].get("processed_text")
+        logger.debug(redacted_text)
+
+        return redacted_text
+
+    def scrub_image(
+        self,
+        image: Image,
+        fill_color: int = config.SCRUB_FILL_COLOR,  # pylint: disable=no-member
+    ) -> Image:
+        """Scrub the image of all PII/PHI.
+
+        Args:
+            image (Image): A PIL.Image object to be scrubbed
+            fill_color (int): The color used to fill the redacted regions(BGR).
+
+        Returns:
+            Image: The scrubbed image with PII and PHI removed.
+        """
+        url = "https://api.private-ai.com/deid/v3/process/files/base64"
+        file_dir = "assets/"
+        file_name = "temp_image_to_scrub.png"
+
+        # save file as "temp_image_to_scrub.png in assets/
+        temp_image_path = os.path.join(file_dir, file_name)
+        image.save(temp_image_path)
+
+        file_type = "image/png"
+
+        # Read from file
+        with open(temp_image_path, "rb") as b64_file:
+            file_data = base64.b64encode(b64_file.read())
+            file_data = file_data.decode("ascii")
+
+        payload = {
+            "file": {"data": file_data, "content_type": file_type},
+            "entity_detection": {"accuracy": "high", "return_entity": True},
+            "pdf_options": {"density": 150, "max_resolution": 2000},
+            "audio_options": {"bleep_start_padding": 0, "bleep_end_padding": 0},
+        }
+
+        headers = {
+            "Content-Type": "application/json",
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(url, json=payload, headers=headers)
+        if response is None:
+            raise ValueError("Private AI request returned None")
+        response = response.json()
+        logger.debug(response)
+        if type(response) is dict and "detail" in response.keys():
```

Comment:
Isn't `type(response.json())` always a dict? If so, can you please remove the unnecessary check?

Reply:
Please add a comment explaining when the response is list and when it's a dict

Reply:
Addressed in: [7a68b77](https://github.com/OpenAdaptAI/OpenAdapt/pull/486/commits/7a68b77c8926486d9e4602d3361e09a721b8f0cb)